### PR TITLE
chore(deps): upgrade some packages (test for node 20)

### PR DESCRIPTION
**Pull request by mattgoud**
### upgrade node modules

Upgrade some packages + audit fix (reduce some security warnings) to prepare for the transition to node 20
 (see [#37041](https://github.com/PrestaShop/PrestaShop/pull/37041) )

| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | upgrade node modules
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | related #34058
| Sponsor company   | Your company or customer's name goes here (if applicable).
| How to test?      | Indicate how to verify that this change works as expected.

### CODE REVIEW

**Comment by ps-jarvis** [comment]
Hello @mattgoud!

This is your first pull request on [hummingbird](https://github.com/PrestaShop/hummingbird/) repository of the PrestaShop project.

Thank you, and welcome to this Open Source community!

File: src/js/components/usePasswordPolicy.ts
Comment (jolelievre) [inline src/js/components/usePasswordPolicy.ts:168]: Forgot this maybe?

**Review by boherm — DISMISSED** [review]

**Review by jolelievre — APPROVED** [review]

File: src/js/components/usePasswordPolicy.ts
Comment (mattgoud) [reply src/js/components/usePasswordPolicy.ts]: ok remove it

**Comment by florine2623** [comment]
No need functional testing :) ok to be merged

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Comment by jolelievre** [comment]
Thanks @mattgoud

**Comment by sneko** [comment]
Just for information, this PR has broken the Storybook setup. But any way it seems it has never been used for years?

**Pull request merged**
